### PR TITLE
Update comments, variable names and assertions on transfer test

### PR DESCRIPTION
**Pull request by bugarela**
Hello :octocat: 

## Description

I've been working a quint specification for ICS20 and, while writing a test based on the `transfer_test.go` test (progress here https://github.com/informalsystems/quint/pull/874), I noticed some of the comments were outdated. Seems like the test was updated to include additional transfer steps, but the comments were not.

This PR:
1. Updates those comments
1. Renames `path` into `pathAtoB` so it works best along with the existing variable named `pathBtoC`
1. Adds an assertion on the balance of chain A at the end (based on outdated comments, felt like this was suppose to be checked)
1. Changes the test on the escrow account to check the denom that was previously in the escrow account, not the default denom. I think this makes more sense than checking for a denom that was never in the escrow account in the first place.
  * PS: I was not able to check that the escrow account actually has any tokens when it is suppose to. I don't know if I'm doing something wrong or if there is something in the test setup that prevents escrow accounts from being updated. 

### Commit Message / Changelog Entry

```bash
test: update comments, variable names and assertions on transfer test
```

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [X] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [X] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [X] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [X] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [X] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/transfer_test.go [hunk 1]
```diff
@@ -39,37 +39,46 @@ func NewTransferPath(chainA, chainB *ibctesting.TestChain) *ibctesting.Path {
...
 
 	// relay send
-	err = path.RelayPacket(packet)
+	err = pathAtoB.RelayPacket(packet)
 	suite.Require().NoError(err) // relay committed
 
+	// check that module account escrow address has locked the tokens
+	escrowAddress := types.GetEscrowAddress(packet.GetSourcePort(), packet.GetSourceChannel())
+	balance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), escrowAddress, sdk.DefaultBondDenom)
+	suite.Require().Equal(coinToSendToB, balance)
```
Comment (crodriguezvega) [inline modules/apps/transfer/transfer_test.go:76]: @bugarela I added the check for the escrow account on chain A. I hope this is what you were trying to add.

**Review by crodriguezvega — APPROVED** [review]
Thanks for these improvement, @bugarela! I added an extra check for the escrow account on chain A. I think one comment needs to be updated, since the balance on sender account on chain A will not be empty, but besides that, the changes look good to me! 👍

File: modules/apps/transfer/transfer_test.go [hunk 2]
```diff
@@ -114,17 +119,20 @@ func (suite *TransferTestSuite) TestHandleMsgTransfer() {
 	err = pathBtoC.RelayPacket(packet)
 	suite.Require().NoError(err) // relay committed
 
-	balance = suite.chainB.GetSimApp().BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), coinSentFromAToB.Denom)
+	// check that balance on chain A is empty
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/transfer/transfer_test.go:125]:
```suggestion
	// check that balance on chain A is updated
```

**Review by damiannolan — APPROVED** [review]
Thank you @bugarela 🙏

File: modules/apps/transfer/transfer_test.go (hunk 1 above)
Comment (bugarela) [reply modules/apps/transfer/transfer_test.go:76]: Thanks! Makes sense to me

**Pull request merged**